jujugui/static/gui/src/app/components/entity-header/entity-header.js

```diff
@@ -39,20 +45,82 @@ YUI.add('entity-header', function() {
       @returns {String} The intial state.
     */
     getInitialState: function() {
-      return {headerHeight: 0};
+      return {
+        headerHeight: 0,
+        modelList: []
+      };
+    },
+
+    componentWillMount: function() {
+      this._fetchModels();
     },
 
     componentDidMount: function() {
       this.setState({headerHeight: this.refs.headerWrapper.clientHeight});
     },
 
+    componentWillUnmount: function() {
+      this.xhrs.forEach((xhr) => {
+        xhr && xhr.abort && xhr.abort();
+      });
+    },
+
+    componentWillReceiveProps: function(nextProps) {
+      // If the user has changed then update the data.
+      if (nextProps.user.user !== this.props.user.user) {
+        this._fetchModels();
+      }
+    },
+
+    /**
+      Makes a request of JEM or JES to fetch the users availble environments.
+
+      @method _fetchModels
+    */
+    _fetchModels: function() {
+      // Delay the call until after the state change to prevent race
+      // conditions.
+      this.setState({loadingModels: true}, () => {
+        var xhr = this.props.listModels(this._fetchModelsCallback);
+        this.xhrs.push(xhr);
```

Comment:
I totally agree, it'd be nice if we could hook into all XHR requests and abort the calls automatically when the component that started them gets unmounted.